espnet/nets/pytorch_backend/e2e_asr_mix_transformer.py

```diff
@@ -0,0 +1,530 @@
+# Copyright 2019 Shigeki Karita
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""Transformer speech recognition model (pytorch)."""
+
+from argparse import Namespace
+import logging
+import math
+
+import numpy
+import torch
+
+from espnet.nets.asr_interface import ASRInterface
+from espnet.nets.ctc_prefix_score import CTCPrefixScore
+from espnet.nets.e2e_asr_common import end_detect
+from espnet.nets.e2e_asr_common import ErrorCalculator
+from espnet.nets.pytorch_backend.ctc import CTC
+from espnet.nets.pytorch_backend.e2e_asr import CTC_LOSS_THRESHOLD
+from espnet.nets.pytorch_backend.e2e_asr import Reporter
+from espnet.nets.pytorch_backend.e2e_asr_mix import E2E as E2EASRMIX
+from espnet.nets.pytorch_backend.e2e_asr_mix import PIT
+from espnet.nets.pytorch_backend.e2e_asr_transformer import E2E as E2EASR
+from espnet.nets.pytorch_backend.nets_utils import get_subsample
+from espnet.nets.pytorch_backend.nets_utils import make_non_pad_mask
+from espnet.nets.pytorch_backend.nets_utils import th_accuracy
+from espnet.nets.pytorch_backend.rnn.decoders import CTC_SCORING_RATIO
+from espnet.nets.pytorch_backend.transformer.add_sos_eos import add_sos_eos
+from espnet.nets.pytorch_backend.transformer.attention import MultiHeadedAttention
+from espnet.nets.pytorch_backend.transformer.decoder import Decoder
+from espnet.nets.pytorch_backend.transformer.encoder import EncoderMix
+from espnet.nets.pytorch_backend.transformer.initializer import initialize
+from espnet.nets.pytorch_backend.transformer.label_smoothing_loss import (
+    LabelSmoothingLoss,  # noqa: H301
+)
+from espnet.nets.pytorch_backend.transformer.mask import subsequent_mask
+from espnet.nets.pytorch_backend.transformer.mask import target_mask
+from espnet.nets.pytorch_backend.transformer.plot import PlotAttentionReport
+
+
+class E2E(ASRInterface, torch.nn.Module):
```

Comment:
Can this E2E inherit `e2e_asr_transformer.E2E` to reuse more redundant methods, e.g., `encode`, plots?

Reply:
Good suggestion, thanks!